docs/custom-images-contract.rst

```diff
@@ -0,0 +1,113 @@
+.. _custom-images-contract:
+
+======================
+Custom Images contract
+======================
+
+The contract is a set of requirements for crawler custom docker images to comply running with Scrapy Cloud.
+
+The contract is already implemented in Python and is known as :ref:`scrapinghub-entrypoint-scrapy <sh-entrypoint-scrapy>`. If you don’t use this package with your crawler image, you should implement Custom Images contract by yourself, or your image will not work properly in Scrapy Cloud.
+
+
+Contract statements
+-------------------
+
+1. Docker image should be able to run via ``start-crawl`` command without arguments.
+
+.. code-block:: bash
+
+    docker run myscrapyimage start-crawl
+
+2. Docker image should be able to return a spiders list via ``list-spiders`` command without arguments.
+
+.. code-block:: bash
+
+   docker run myscrapyimage list-spiders
+
+3. Crawler should be able to get all needed params using :ref:`system environment variables <environment-variables>`.
+
+
+.. _environment-variables:
+
+Environment variables
+---------------------
+
+SHUB_JOBKEY
+^^^^^^^^^^^
+
+Job key in format ``PROJECT_ID/SPIDER_ID/JOB_ID``.
+
+**Example**:
+
+.. code-block:: javascript
+
+    123/45/67
+
+
+SHUB_JOB_DATA
+^^^^^^^^^^^^^
+
+Job arguments, in json format.
+
+**Example**:
+
+.. code-block:: javascript
+
+    {"key": "1111112/2/2", "project": 1111112, "version": "version1",
+    "spider": "spider-name", "spider_type": "auto", "tags": [],
+    "priority": 2, "scheduled_by": "user", "started_by": "admin",
+    "pending_time": 1460374516193, "running_time": 1460374557448, ... }
+
+SHUB_JOB_SETTINGS
```

Comment:
> is it too late

we don't pass a variable with name `SHUB_JOB_SETTINGS` to the job, so it's not late. @vshlapakov needs to add these variables before we merge this PR though. I almost missed that.